src/input/mod.rs

```diff
@@ -1833,6 +1841,15 @@ impl State {
             self.niri.pointer_hidden = false;
             self.niri.tablet_cursor_location = None;
 
+            self.niri.pick_window.retain(|tx| !tx.is_closed());
+            if !self.niri.pick_window.is_empty() {
+                if let Some(mapped_id) = self.niri.window_under_cursor().map(|mapped| mapped.id()) {
```

Comment:
Could probably fetch `window_under_cursor()` above to reuse it also in the interactive move condition right below and not fetch twice